nds-org/kubeadm-terraform
# NFS Provisioner Support

# Problem
Fixes https://github.com/nds-org/kubeadm-terraform/issues/2, which describes the desire to be able to dynamically provision PersistentVolumes in Kubernetes using NFS.

# Approach

Adds support for the [NFS Provisioner](https://github.com/kubernetes-incubator/external-storage/tree/master/nfs) under the right conditions:
* When `storage_node_count == 0`, no PVC support will be included by default
* When `storage_node_count >= 1`, the NFS provisioner will be deployed

# How to test

1. Edit `variables.tf` and adjust `storage_node_count` to 1
2. Execute `terraform init && terraform apply`
3. Wait for your cluster to come up
4. SSH into the master node
5. Execute `kubectl create -f nfs/test.pvc.yaml`
6. Execute `kubectl get pvc,pv`
    * After a few seconds, you should see that your PVC has caused the creation of a new PV
